tests/02_eradiate/01_unit/scenes/atmosphere/test_heterogeneous.py

```diff
@@ -306,11 +313,92 @@ def test_heterogeneous_blend_switches(mode_mono):
         particle_layers=[ParticleLayer()],
     )
 
-    # Purely absorbing atmosphere + particle layer combination is not allowed
-    with pytest.raises(ValueError):
-        HeterogeneousAtmosphere(
-            molecular_atmosphere=MolecularAtmosphere.ussa_1976(
-                has_absorption=True, has_scattering=False
+
+def test_heterogeneous_absorbing_mol_atm(
+    mode_ckd, tmp_path, absorbing_only, scattering_only
+):
+    """
+    Phase function weights are correct when the molecular atmosphere is
+    absorbing-only and the particle layer is either absorbing-only or
+    scattering-only.
+    """
+    # Create the heterogeneous atmosphere
+    pl_bottom = 1.0 * ureg.km  # arbitrary
+    pl_top = 4.0 * ureg.km  # arbitrary
+    particle_layer = ParticleLayer(
+        bottom=pl_bottom,
+        top=pl_top,
+        dataset=scattering_only,
+    )
+    atmosphere1 = HeterogeneousAtmosphere(
+        molecular_atmosphere=MolecularAtmosphere.afgl_1986(
+            has_absorption=True,
+            has_scattering=False,
+        ),
+        particle_layers=particle_layer,
+        geometry={"type": "spherical_shell"},
+    )
+    atmosphere2 = attrs.evolve(
+        atmosphere1,
+        particle_layers=ParticleLayer(
+            bottom=pl_bottom,
+            top=pl_top,
+            dataset=absorbing_only,
+        ),
+    )
+
+    # Generate kernel dict
+    spectral_ctx = CKDSpectralContext(bin_set="1nm")
+    ctx = KernelDictContext(
+        spectral_ctx=spectral_ctx,
+    )
+    kernel_dict1 = atmosphere1.kernel_phase(ctx)
+    kernel_dict2 = atmosphere2.kernel_phase(ctx)
+
+    # Write phase function weight data to disk
+    weight_data1 = kernel_dict1.data["phase"]["weight"]["grid"]
+    path1 = tmp_path / "gridvolume1.vol"
+    weight_data1.write(str(path1))
+    weight_data2 = kernel_dict2.data["phase"]["weight"]["grid"]
+    path2 = tmp_path / "gridvolume2.vol"
+    weight_data2.write(str(path2))
+
+    # Read the phase function weight data from disk
+    weight1 = read_binary_grid3d(path1)
+    z_units = "km"
+    weight1 = xr.DataArray(
```

Comment:
I don't understand the point of constructing data arrays to perform the comparison: since you're using the same kernel units, you should be able to compare unitless, unlabelled arrays.